packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -0,0 +1,55 @@
+import * as tsserver from 'typescript/lib/tsserverlibrary';
+
+const doNothing = (): void => {};
+
+const createStubFileWatcher = (): tsserver.FileWatcher => ({
+  close: doNothing,
+});
+
+export function createProjectService(): tsserver.server.ProjectService {
+  // TODO: see getWatchProgramsForProjects
+  // We don't watch the disk, we just refer to these when ESLint calls us
+  // there's a whole separate update pass in maybeInvalidateProgram at the bottom of getWatchProgramsForProjects
+  // (this "goes nuclear on TypeScript")
+  const system: tsserver.server.ServerHost = {
+    ...tsserver.sys,
+    clearImmediate,
+    clearTimeout,
+    setImmediate,
+    setTimeout,
```

Comment:
These are already provided by `ts.sys`, technically; tsserver actually just does `sys as ServerHost`, which is unsafe, but, seemingly okay if you want to write less here.

Reply:
Hmm, that assertion does make me nervous. Do you think it'd be reasonable to indicate that `ts.sys` is a `System` with those properties filled out?

Reply:
It's tricky because outside of node, those may not actually exist. It's fine to leave it as-is, I think.